txmongo/_gridfs/__init__.py

```diff
@@ -62,6 +62,9 @@ def __init__(self, database, collection="fs"):
         self.__indexes_created_defer = defer.DeferredList([
             self.__files.create_index(
                 filter.sort(ASCENDING("filesname") + ASCENDING("uploadDate"))),
+            self.__files.create_index(
+                filter.sort(ASCENDING("filesname") + DESCENDING("uploadDate"))),
```

Comment:
Ok, in my defence I was just copying other people's code with regards to index creation, and if one of those indexes is obsolete, it would imply the current method for specifying index creation is also rather redundant. I've done a little reading too now ( :) ) so here's another shot with one index, with a slightly more direct specification, and also I modified the test to it runs under both Python2 "and" Python3.